### ISSUE TITLE
deprecate `rocks.api.source_runtime_dir` -> Extract to its own library

### ISSUE THREAD

**Issue by mrcjkb** [nvim-neorocks/rocks.nvim#294]
The logic for sourcing runtime directories is no longer needed by rocks.nvim itself (after #292).

It is still used by rocks-git.nvim.

We should extract it to its own luarocks library, and deprecate it for removal in rocks.nvim 3.0.0.

**Comment by mrcjkb** [comment]
#297
